transport/peers/peer.go

```diff
@@ -0,0 +1,82 @@
+package peers
+
+import (
+	"go.uber.org/yarpc/transport"
+
+	"github.com/uber-go/atomic"
+)
+
+// NewPeerIdentifier creates a new HostPortPeerIdentifier
+func NewPeerIdentifier(hostport string) *HostPortPeerIdentifier {
+	return &HostPortPeerIdentifier{
+		hostport: hostport,
+	}
+}
+
+// HostPortPeerIdentifier uniquely references a host:port combination using a common interface
+type HostPortPeerIdentifier struct {
+	hostport string
+}
+
+// Identifier generates a (should be) unique identifier for this PeerIdentifier (to use in maps, etc)
+func (p *HostPortPeerIdentifier) Identifier() string {
+	return p.hostport
+}
+
+// NewPeer creates a new HostPortPeer from a PeerIdentifier, Agent, and Subscriber
+func NewPeer(pid transport.PeerIdentifier, agent transport.PeerAgent, subscriber transport.PeerSubscriber) *HostPortPeer {
+	hppid := pid.(*HostPortPeerIdentifier)
+	return &HostPortPeer{
+		HostPortPeerIdentifier: HostPortPeerIdentifier{
+			hostport: hppid.hostport,
+		},
+		agent:            agent,
+		subscriber:       subscriber,
+		pending:          atomic.NewInt32(0),
+		connectionStatus: transport.PeerAvailable,
+	}
+}
+
+// HostPortPeer keeps a subscriber to send status updates to it, and the PeerAgent that created it
+type HostPortPeer struct {
+	HostPortPeerIdentifier
+
+	agent            transport.PeerAgent
+	subscriber       transport.PeerSubscriber
+	pending          *atomic.Int32
```

Comment:
You can make `atomic.Int32` a value rather than a pointer and then you don't
have to initialize it in `NewPeer`. The atomic operations will still work as
long as `HostPortPeer` is a pointer.